File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        if (nums1.size() > nums2.size()) swap(nums1, nums2);
        unordered_set<int> s(nums1.begin(), nums1.end());
        vector<int> result;
        for (int num : nums2) {
            if (s.erase(num)) { 
                result.push_back(num);
            }
        }    
        return result;
    }
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1.sort()
        nums2.sort()
        res=[]
        i,j,x,y=0,0,len(nums1),len(nums2)
        while i<x and j<y:
            if nums1[i]==nums2[j] and (len(res)==0 or res[-1]!=nums1[i]):
                res.append(nums1[i])
            elif nums1[i]<nums2[j]:
                i+=1
            else:
                j+=1
        return res